# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/CevdetSatarr/FlyRank-intern/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

I have chosen **Lane 2: Refresh / Content Opportunity Scoring**.

This lane focuses on identifying which specific content items or pages should be prioritized for review based on observable historical data. It is highly worth our next 7 weeks because content management is constrained by human capacity. Instead of manually auditing thousands of pages, a data-driven opportunity score ensures that the content review team spends their limited hours on the most promising candidates first, directly protecting traffic and capturing hidden search demand.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.



*   What decision does this improve?
    It improves the decision of which pages a human reviewer or editor should inspect and update first out of the entire content inventory.
    
*   Who acts on the output, and what do they do?
    The content optimization team or editorial reviewers act on the output. They will take the top-ranked pages from the generated queue, inspect their specific reason codes, and execute real-time on-page improvements like updating stale information, expanding thin content, or fixing meta descriptions.
    
*   What does a wrong answer cost?
    - False Positives (High score given to a page that doesn't need a fix or won't recover): Wastes valuable editor hours and operational capacity on low-impact tasks.
    - False Negatives (Missing a critical page that is decaying): Costs missed search impressions, traffic loss to competitors, and permanent erosion of domain authority.
    
*   Why does data or ML help at all?
    Simple heuristic rules (like refresh if a page is older than 180 days) are too rigid and completely ignore actual search intent, position tiers, and multi-dimensional traffic trends. User search patterns change dynamically, and traffic shifts across many tangled signals like impressions, clicks, CTR, and average position. Machine learning can synthesize these complex historical interactions far better than static, manual SQL rules.

### The One-Paragraph Frame
For the content editorial team, deciding which pages to review and refresh first, we will build a ranking and scoring model from the anonymized starter dataset (content_refresh_anonymized.csv), predicting an opportunity or decline score measured by Precision@K and Average Precision. A wrong call costs wasted editor hours or missed traffic recovery. A plain rule isn't enough because search demand, content decay, and user engagement signals are highly non-linear and change dynamically over time. We will claim only decision-support results to help humans optimize their workflow.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load the anonymized starter dataset
df = pd.read_csv("/content/content_refresh_anonymized.csv")

# 1. Total Rows
total_rows = len(df)

# 2. Target Distribution (Starter target is trend_direction == 'down')
decline_count = (df['trend_direction'] == 'down').sum()
decline_pct = (df['trend_direction'] == 'down').mean() * 100

print(f"Total Rows: {total_rows}")
print(f"Declining Pages: {decline_count} ({decline_pct:.2f}%)")

Total Rows: 30000
Declining Pages: 16262 (54.21%)



A quick analysis of the `data/raw/content_refresh_anonymized.csv` starter dataset yields the following baseline metrics:

* **Dataset Size:** 30,000 unique pseudonymized content rows.
* **Class Imbalance / Base Rate:** 54.21% of the pages are labeled as `trend_direction == 'down'`.

This high base rate confirms a widespread pattern of content decay across the monitored pages. Because over half of the inventory shows a downward trend, human editors cannot review everything at once. This capacity constraint justifies building a ranking model evaluated via Precision@K (e.g., Precision@20 or Precision@50) to filter and surface only the highest-impact optimization opportunities first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


*   **What I CAN claim:** I can claim that our model can effectively rank and prioritize content assets based on observable historical signs of decay or opportunity, ensuring human review capacity is directed toward high-probability candidates.
*   **What I CANNOT claim:** I cannot claim that our model proves Google algorithm factors or guarantees that a refresh will definitely cause a traffic recovery. This dataset only tracks observational search metrics; proving causality would require a separate experimental or causal design.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.